src/storage/admin/AdminTaskManager.cpp

```diff
@@ -85,6 +159,27 @@ void AdminTaskManager::shutdown() {
   LOG(INFO) << "exit AdminTaskManager::shutdown()";
 }
 
+void AdminTaskManager::saveTaskStatus(JobID jobId,
+                                      TaskID taskId,
+                                      meta::cpp2::AdminCmd cmd,
+                                      nebula::meta::cpp2::JobStatus jobStatus) {
+  if (env_ == nullptr) return;
+  std::string key = NebulaKeyUtils::adminTaskKey(env_->adminSeqId_, jobId, taskId);
+  std::string val;
+  val.reserve(sizeof(nebula::meta::cpp2::JobStatus) + sizeof(meta::cpp2::AdminCmd));
+  val.append(reinterpret_cast<char*>(&cmd), sizeof(meta::cpp2::AdminCmd));
+  val.append(reinterpret_cast<char*>(&jobStatus), sizeof(nebula::meta::cpp2::JobStatus));
+  env_->adminStore_->put(key, val);
+  env_->adminStore_->flush();
+}
+
+void AdminTaskManager::removeTaskStatus(JobID jobId, TaskID taskId) {
+  if (env_ == nullptr) return;
+  std::string key = NebulaKeyUtils::adminTaskKey(env_->adminSeqId_, jobId, taskId);
```

Comment:
Why we need a `adminSeqId_`? `jobId + taskId` will make sure unique

Reply:
> make

I should avoid to report tasks in the current term.

Reply:
👌 , I got it.